### ISSUE TITLE
Logic error with precedence

### ISSUE THREAD

**Issue by nkavian** [PebbleTemplates/pebble#205]
The two statements below are logically the same.  Pebble only displays XYZ.

```
{% set a = false %}
{% set b = true %}
{% if not a and not b %}
  XYZ
{% endif %}
{% if (not a) and (not b) %}
  123
{% endif %}
```

I think Pebble is evaluating `{% if not a and not b %}` as if it were `{% if not (a and not b) %}`

`not` should take higher precedence over other things like `and`.

**Comment by mbosecke** [comment]
Out of curiosity, I upped the precedence of the `not` operator and ran all of our unit tests. It broke a bunch of tests because this is a common way of using it:

`{% if not 2 is odd %}`
...
`{% endif %}`

I.e. it's used to negate a much larger boolean expression. In my opinion, both use cases seem reasonable but I think I like yours a little better. Unfortunately, it was a design decision made back in 2014 so it would break a lot of templates to change it at this point.

I'll keep this task open to hear more opinions and have it considered for the next major release of Pebble when we're willing to break backwards compatibility. This may not be for a long time.

**Comment by mbosecke** [comment]
Implemented on the 3.0 branch
